<launch>
  <!-- declare arg to be passed in -->
  <arg name="do_remote" default="0" /> 
  <arg name="machine" default="" />
    
  <!-- if set to run remotely, load remote machine config -->
  <group if="$(arg do_remote)">
    <include file="$(find ni_depth_segmentation)/launch/$(arg machine).machine" />
  </group>
  
  <group if="$(arg do_remote)">
    <node machine="nivision" pkg="ni_depth_segmentation" name="attention" type="attention_node" output="screen">
      <param name="path_color" type="str" value="$(arg path_color)"/>
      <param name="path_sift" type="str" value="$(arg path_sift)"/>
    </node>
  </group>
  <group unless="$(arg do_remote)">
    <node pkg="ni_depth_segmentation" name="attention" type="attention_node" output="screen">
      <param name="path_color" type="str" value="$(arg path_color)"/>
      <param name="path_sift" type="str" value="$(arg path_sift)"/>
    </node>
  </group>

</launch>
